tsl/src/compression/api.c

```diff
@@ -240,6 +240,23 @@ compress_chunk_impl(Oid hypertable_relid, Oid chunk_relid)
 					AccessShareLock);
 	LockRelationOid(catalog_get_table_id(ts_catalog_get(), CHUNK), RowExclusiveLock);
 
+	/*
+	 * Re-read the state of the chunk after all locks have been acquired and ensure
+	 * it is still uncompressed. Another process running in parallel might have
+	 * already performed compression while we were waiting for the locks to be
+	 * acquired.
+	 */
+	Chunk *chunk_state_after_lock = ts_chunk_get_by_relid(chunk_relid, true);
+	if (chunk_state_after_lock->fd.compressed_chunk_id != INVALID_CHUNK_ID)
+	{
+		ts_cache_release(hcache);
+		ereport(NOTICE,
+				(errcode(ERRCODE_DUPLICATE_OBJECT),
```

Comment:
I don't think this is a good error code for this case (it seems `ERRCODE_DUPLICATE_OBJECT` is used when there is an attempt to redefine an already existing object, which I do not really think is the case here since we treat the operation as idempotent) but since this is a notice, I think you can just remove the error code.

Reply:
We also use `ERRCODE_DUPLICATE_OBJECT` in other places (e.g., Line 383, 440) to indicate that a chunk is already compressed. I am in favor of returning the same error code for the same error in all places. Are you in favor of changing the error code everywhere?

Reply:
That is a good point. It is better to be consistent than to have the perfect error code. You can leave it as it is.